src/librustdoc/core.rs

```diff
@@ -273,6 +274,58 @@ where
     (lint_opts, lint_caps)
 }
 
+crate fn get_rustdoc_lints(
+    cfg: FxHashSet<(String, Option<String>)>,
+    opts: config::Options,
+) -> Vec<String> {
+    let config = interface::Config {
+        opts,
+        crate_cfg: cfg,
+        input: Input::File(PathBuf::new()),
+        input_path: None,
+        output_file: None,
+        output_dir: None,
+        file_loader: None,
+        diagnostic_output: DiagnosticOutput::Default,
+        stderr: None,
+        lint_caps: Default::default(),
+        register_lints: None,
+        override_queries: None,
+        make_codegen_backend: None,
+        registry: rustc_driver::diagnostics_registry(),
+    };
+    let missing_docs_lints = rustc_lint::builtin::MISSING_DOCS.name;
+    let renamed_and_removed_lints = rustc_lint::builtin::RENAMED_AND_REMOVED_LINTS.name;
+    let unknown_lints = rustc_lint::builtin::UNKNOWN_LINTS.name;
+
+    // In addition to those specific lints, we also need to allow those given through
+    // command line, otherwise they'll get ignored and we don't want that.
+    let mut rustdoc_lints = vec![
+        missing_docs_lints.to_owned(),
+        renamed_and_removed_lints.to_owned(),
+        unknown_lints.to_owned(),
+    ];
+
+    interface::run_compiler(config, |compiler| {
+        let sopts = &compiler.session().opts;
+        let lint_store = rustc_lint::new_lint_store(
+            sopts.debugging_opts.no_interleave_lints,
+            compiler.session().unstable_options(),
+        );
+        if let Some((_, lints, _)) =
+            lint_store.get_lint_groups().iter().find(|(name, _, _)| *name == "rustdoc")
+        {
+            for lint in lints {
+                // Funny thing about this: the lints in the compiler are capitalized but the lints
+                // you get through the `LintStore` are not, which means you have to capitalize them.
+                rustdoc_lints.push(lint.to_string().to_uppercase());
```

Comment:
Does this work?
```suggestion
                rustdoc_lints.push(lint.name);
```
I think it would let you get rid of the `to_owned()` on line 304 too.

Reply:
With the two suggestions:

```
   Compiling rustdoc v0.0.0 (/home/imperio/rust/rust/src/librustdoc)
error[E0609]: no field `name` on type `&rustc_lint::LintId`
   --> src/librustdoc/core.rs:321:41
    |
321 |                 rustdoc_lints.push(lint.name);
    |                                         ^^^^ unknown field
    |
    = note: available fields are: `lint`

error[E0308]: mismatched types
   --> src/librustdoc/core.rs:326:5
    |
280 | ) -> Vec<String> {
    |      ----------- expected `Vec<std::string::String>` because of return type
...
326 |     rustdoc_lints
    |     ^^^^^^^^^^^^^ expected struct `std::string::String`, found `&str`
    |
    = note: expected struct `Vec<std::string::String>`
               found struct `Vec<&str>`

error: aborting due to 2 previous errors
```

So I'll keep things as is. ;)